agent/hcp/client/client.go

```diff
@@ -14,6 +14,7 @@ import (
 
 	"github.com/hashicorp/consul/agent/hcp/config"
 	"github.com/hashicorp/consul/version"
+	hcptelemetry "github.com/hashicorp/hcp-sdk-go/clients/cloud-consul-telemetry-gateway/preview/2023-04-14/client/consul_telemetry_service"
```

Comment:
Update the Public Go HCP SDK to get the new FetchTelemetryConfig endpoint that returns the CCM Telemetry Config.